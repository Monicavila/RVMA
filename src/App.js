import React from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import Landing from "./components/landing/landing";
import Meeting from "./components/createRoom/createRoom";
import Room from "./components/room/room";

function App() {
  return (
    <BrowserRouter>
      <Switch>
        <Route path="/" exact component={Meeting} />
        <Route path="/room/:roomID" render={(props) => {const { match } = props; return <Room {...props} roomID={match.params.roomID} />}}/>
        <Route path="/back-soon" exact component={Landing} />
      </Switch>
    </BrowserRouter>
  );
}

export default App;