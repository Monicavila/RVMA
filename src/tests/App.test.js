import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from "react-dom/test-utils";
import App from '../App';

test('renders learn react link', () => {
  act(()=>{
    render(<App />);
  })
  const linkElement = screen.getByText("NEW MEETING");
  expect(linkElement).toBeInTheDocument();
});

it('renders a snapshot', () => {
  const tree = renderer.create(<App/>).toJSON();
  expect(tree).toMatchSnapshot();
});