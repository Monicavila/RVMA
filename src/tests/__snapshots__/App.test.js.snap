// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="containerCR"
>
  <div
    className="creatingRoom"
  >
    <div
      className="left_part"
    >
      <div
        className="on"
      />
    </div>
    <div
      className="centerDown"
    >
      <div
        className="slider"
      >
        <div
          className="switch_button"
        >
          <input
            className="switch_button__checkbox"
            id="switch_label"
            name="switch_button"
            onClick={[Function]}
            type="checkbox"
          />
          <label
            className="switch_button__label"
            htmlFor="switch_label"
          />
        </div>
      </div>
    </div>
    <div
      className="right_part"
    >
      <div
        className="logobla"
      />
      <div
        className="letsnack"
      >
        <p
          className="snacktogether"
        >
          Let's snack together while meeting is going on.
          <br />
          <br />
          It's FREE! Try it!
        </p>
      </div>
      <div
        className="newmeet"
        onClick={[Function]}
      >
        NEW MEETING
      </div>
      <input
        className="link_input"
        onChange={[Function]}
        placeholder="Enter your URL"
        type="text"
      />
      <div
        className="join"
        onClick={[Function]}
        path="/"
      >
        Join!
      </div>
    </div>
  </div>
</div>
`;
